hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -120,7 +120,7 @@ object InferPType {
     }
     node._pType = node match {
       case x if x.typ == TVoid => PVoid
-      case _: I32 | _: I64 | _: F32 | _: F64 | _: Str | _: UUID4 | _: Literal | _: True | _: False
+      case _: I32 | _: I64 | _: F32 | _: F64 | _: Str | _: UUID4 | _: Literal | _: EncodedLiteral | _: True | _: False
```

Comment:
encodedLiteral should have its own rule that calls cidec.decodedPType(), even if the two will usually be the same. Should add an assertion in the emitter that the decodedPType() is equal to the ptype of the node.